 <div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom">
    <a href="https://www.mathjax.org">
    <img style="height:1.5em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" >
</a></span></div>

<div style="display:inline-block; position:absolute; top:0; left:0; color: #230e3e; font:italic 1em serif;"><a href="https://www.julialang.org">
    <img style="height:2.25em; vertical-align:top" title="Powered by Julia"
    src="./figures/julia-logo-mask.png"
    border="0" alt="Powered by Julia" >
</a></div>
</a></div>$\require{cancel} \require{mhchem} \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols}$

In [1]:
using Unitful
using Printf

<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 1em" >&nbsp;</p>

GOLD **R-134a at 1000 kPa is throttled to a temperature of –16 °C and a quality of 100%. The velocity at the inlet and outlet is 10 m/s. Find:**   <ol type="a">
1. **Inlet quality**  
2. **Ratio of inlet area to outlet area** </ol>

### Get Organized and Attack

We don't know state ➀, but we know state ➁, and we know **throttles are isenthalpic** and that we have steady state conditions (the system isn't accumulating mass or energy). 

#### 1. Find $x_1$  
<ol>
  All we need to do is resolve state ➀ and calculate it's quality.  $$$$
    
$
\text{R-134a}\left|\small \begin{array}{l|l}  
 T_2 = -16\,\ce{°C} & P_1 = 1\, \ce{MPa} \\[-3pt]
 x_2:\text{sat vapor} &  h_1 = h_2 & \end{array} \right. \\[12pt]
$

In [2]:
h2 = 237.74u"kJ/kg"           # R-134a{T = -16 °C, sat vapor} 
h1 = h2                       # it's a throttle

h1f = 105.29u"kJ/kg"          # sat liquid @ P1 = 1 MPa
h1g = 267.97u"kJ/kg"          # sat vapor @ P1 = 1 MPa

# h1f < h1 < h1g:  we know it is saturated. We are in the right place.

x1 = (h2 - h1f)/(h1g - h1f)   # calculate quality
println("x1 = ",x1)

x1 = 0.8141750676174083


<div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize x_1 = 0.814 $$
</div></div>


#### 2. Find $\displaystyle \frac{A_{in}}{A_{out}}$  
<ol>
  Conservation of mass is the key here. Resolved into area and velocity, $\dot m = \rho \vec V A.$ With one inlet, one outlet, at steady state,</ol>    
   
\begin{align*}
 \dot m_{in} &= \dot m_{out} \\[6pt]
 \frac{\vec{V}_1A_1}{v_1} &= \frac{\vec{V}_2 A_2}{v_2} \\
\end{align*}

<ol>Now, $ \vec{V}_1 = \vec{V}_2,$ so 
$$
\frac{A_1}{A_2} = \frac{v_2}{v_1}
$$


In [3]:
v2 = 0.1247u"m^3/kJ"           # R-134a{T = -16 °C, sat vapor} 

# need to resolve v1
# remember, we already know x1

v1f = 1.1416e-3u"m^3/kJ"       # sat liquid @ P1 = 1 MPa
v1g = 0.0053u"m^3/kJ"          # sat vapor @ P1 = 1 MPa

v1 = v1f + x1 * (v1g - v1f)    # calculate v1
@printf("A1 / A2 = %.4f",v2/v1)

A1 / A2 = 27.5442

#### <div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize \frac{A_1}{A_2} = 27.54 $$
</div></div>

<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 1em" >&nbsp;</p>

**An air-conditioner with an average COP of 2.2 consumes 36 kWh of electricity during a certain day. Calculate the amount of heat removed by this air-conditioner that day.** 

$$ \sf \text{COP}_R = \frac{\text{desired output}}{\text{required input}} = \frac{Q_L}{W_{in}} 
$$


In [4]:
COP = 2.2
Qʟ = COP * 36u"kW*hr" |> u"kJ"

285120.0 kJ

<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 1em" >&nbsp;</p>

VIRIDIS **A 1000-L tank initially contains water at 300 kPa and a quality of 1%. Heat is transferred to the water, thereby raising its pressure and temperature. At a pressure of 2.5 MPa, a safety valve opens and saturated vapor at 2.5 MPa flows out. The process continues, maintaining 2.5 MPa until the quality in the tank is 90%, then stops. Determine the mass of water to exit and the total heat transfer (with direction).** 

![steamdrum](./figures/steamdrum.png.pdf)


In [5]:
# Given
V = 1000u"L"
x1 = 0.01
x3 = 0.9;

Organization is key for a problem like this. However, the solution method is the same as every other process problem we've solved: start with what you know, consider what you need, and plot the path one process at a time. 

A few notes:
- unsteady flow
- only saturated vapor escapes
- negligible potential or kinetic energy effects

### Get Organized and Attack



Here's what we're given, arranged by state and process. We will apply conservation of mass and conservation of energy for both processes.   

 $$1000\, \ce{L} \ \ce{H2O} \left| 
 \begin{array}{ll} 
  P_{1} = 300\, \ce{kPa} \\
  x_{1} = 1\% \\
 \end{array}\right|
 \xrightarrow{isochoric}
 \left| 
 \begin{array}{ll} 
   P_2 = 2.5\, \ce{MPa} \\
   v_2 = v_1 \\
 \end{array}\right|
 \xrightarrow{isobaric}
 \left| 
 \begin{array}{ll} 
   P_3 = 2.5\, \ce{MPa} \\
   x_3 = 90\% \\
 \end{array}\right|
 $$

<p style = "width:100vw; background-image:  url(./figures/bacgroundTiles.png); line-height: 0.35em" >&nbsp;</p>

### Isochoric Heating from State_1 to State_2 

##### Mass  
Our system is closed for this process. Nothing to do here but calculkate the mass.  

$\qquad m_1 = m_2 = \displaystyle \frac{V}{v_1}$

##### Energy    
$\qquad
\begin{array}{lcccl}
  E_{in} &-& E_{out} &=& \Delta E_{sys} \\[6pt]
  Q_{in,1\rightarrow2} &&  &=& U_2 - U_1 = m_2u_2 - m_1u_1\\[6pt]
\end{array}  
$  
$$$$   
   
$\qquad Q_{in,1\rightarrow2} = m_1 \left(u_2 - u_1 \right) $

In [6]:
# Get State 1 ⋮ Given P1 = 300 kPa, x3 = 0.01
vf1 = 1.0732e-3u"m^3/kg"
vg1 = 0.6058u"m^3/kg"
uf1 = 561.15u"kJ/kg"
ug1 = 2543.6u"kJ/kg"

v1 = vf1 + x1 * (vg1 - vf1)
u1 = uf1 + x1 * (ug1 - uf1)
# @printf("v1 = %s",v1); println()
# @printf("u1 = %s",u1); println()

580.9744999999999 kJ kg^-1

In [7]:
# Get State 2 ⋮ Given P2 = 2.5 MPa, v2 = v1
P2 = 2.5u"MPa"
v2 = v1

vf2 = 1.1973e-3u"m^3/kg"
vg2 = 0.07998u"m^3/kg"
uf2 = 959.11u"kJ/kg"
ug2 = 2603.1u"kJ/kg"

x2 = (v2 - vf2)/(vg2 - vf2)
u2 = uf2 + x2 * (ug2 - uf2)

m2 = V/v2 |> u"kg"
@printf("m2 = %s",m2); println()

# Energy Equation
Qin12 = m2 * (u2 - u1)

@printf("Qin1→2 = %s",Qin12); println()

m2 = 140.44020702010036 kg
Qin1→2 = 70463.99300632655 kJ


<p style = "width:100vw; background-image:  url(./figures/bacgroundTiles.png); line-height: 0.35em" >&nbsp;</p>

### Isobaric Heating from State_2 to State_3 


##### Mass  
Now the valve is open and we have an unsteady flow, open system for the duration of the process. 

$\qquad 
\begin{array}{lcrcl}
  m_{in} &-& m_{out} &=& \Delta m_{sys}  \\[6pt]    
  &-& m_{out,2\rightarrow3}  &=& m_3 - m_2 \\[18pt]
  \hline
  && m_{out,2\rightarrow3} &=& m_2 - m_3
\end{array} 
$

##### Energy    
Now is a good time to remind yourself that all the $\ce{H2O}$ escaping is saturated vapor at tank pressure.  

$\qquad
\begin{array}{lcrcl}
  E_{in} &-& E_{out} &=& \Delta E_{sys} \\[6pt]
  Q_{in,2\rightarrow3} &-& m_{out,2\rightarrow3}h_{g,3} &=& m_3u_3 - m_2u_2\\[18pt]
  \hline
  && Q_{in,2\rightarrow3} &=& m_3u_3 - m_2u_2 + m_{out,2\rightarrow3}h_{g,3}
\end{array}  
$  

In [8]:
# Get State 3 ⋮ Given P3 = 2.5 MPa, x3 = 90%
# We're still saturated at P3 = P2, so
# many propoerties will be the same
P3 = P2
x3 = 0.9

vf3 = vf2
vg3 = vg2
uf3 = uf2
ug3 = ug2

hg3 = 2803.1u"kJ/kg"   # the escaping steam

v3 = vf3 + x3 * (vg3 - vf3)
u3 = uf3 + x3 * (ug3 - uf3)

m3 = V/v3
m_out = m2 - m3

# Energy equation
Qin23 = m3*u3 - m2*u2 + m_out*hg3; 

### A little summary algebra and you're home


In [9]:
m_out = m2 - m3
Qin = Qin12 + Qin23 |> u"kJ"

# @printf("Q23in = %s",Q23in); println()
@printf("Qin = %s",Qin); println()
@printf("m_out = %s",m_out);

Qin = 307021.8088416562 kJ
m_out = 126.57091428607026 kg

#### <div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize Q_{in} = 307.02\ \ce{MJ} \qquad \dot m_{out} = 126.57\ \ce{kg}$$
</div></div>

<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 1em" >&nbsp;</p>

**A proposed power cycle (see figure below) is to have a thermal efficiency of 30% while receiving energy by heat transfer from condensing steam and discharging energy by heat transfer to a nearby lake at 20 °C. Determine the lowest possible condensing steam pressure (highest possible cycle efficiency).**

The key expression here is that we were asking for the "best" of something. That should kake you think reversible processes, and that 

$$\left(\frac{Q_L}{Q_H}\right)_{rev} = \frac{T_L}{T_H}$$

From there, 

$$\eta_{th, max} = 1 - \left( \frac{Q_L}{Q_H} \right)_{rev} = 1 - \frac{T_L}{T_H}
$$   

$$$$

$$ T_H = \frac{T_L}{1 - \eta_{th, max}}
$$

In [10]:
ηth = 0.3
Tʟ = 20u"°C" |> u"K"
Tʜ = Tʟ / (1 - ηth) |> u"°C"
# @printf('Tʜ ='')

145.6357142857143 °C

With that temperature, we need the corresponding saturation pressure for our condensing steam:  $$$$
<div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize P = P_{sat@Tʜ} = 432\, \ce{kPa} $$
</div></div>

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>